plugins/inventory/proxmox.py

```diff
@@ -370,109 +363,132 @@ def to_safe(self, word):
-            stopped_group = 'all_stopped'
-            stopped_group = self.to_safe('%s%s' % (self.get_option('group_prefix'), stopped_group.lower()))
-            self.inventory.add_group(stopped_group)
-
-            if node.get('node'):
-                self.inventory.add_host(node['node'])
-
-                if node['type'] == 'node':
-                    self.inventory.add_child(nodes_group, node['node'])
-
-                if node['status'] == 'offline':
-                    continue
-
-                # get node IP address
-                if self.get_option("want_proxmox_nodes_ansible_host"):
-                    ip = self._get_node_ip(node['node'])
-                    self.inventory.set_variable(node['node'], 'ansible_host', ip)
-
-                # get LXC containers for this node
-                node_lxc_group = self.to_safe('%s%s' % (self.get_option('group_prefix'), ('%s_lxc' % node['node']).lower()))
-                self.inventory.add_group(node_lxc_group)
-                for lxc in self._get_lxc_per_node(node['node']):
-                    self.inventory.add_host(lxc['name'])
-                    self.inventory.add_child(lxc_group, lxc['name'])
-                    self.inventory.add_child(node_lxc_group, lxc['name'])
-
-                    # get LXC status when want_facts == True
-                    if self.get_option('want_facts'):
-                        self._get_vm_status(node['node'], lxc['vmid'], 'lxc', lxc['name'])
-                        if lxc['status'] == 'stopped':
-                            self.inventory.add_child(stopped_group, lxc['name'])
-                        elif lxc['status'] == 'running':
-                            self.inventory.add_child(running_group, lxc['name'])
-
-                    # get LXC config and snapshots for facts
-                    if self.get_option('want_facts'):
-                        self._get_vm_config(node['node'], lxc['vmid'], 'lxc', lxc['name'])
-                        self._get_vm_snapshots(node['node'], lxc['vmid'], 'lxc', lxc['name'])
-
-                    self._apply_constructable(lxc["name"], self.inventory.get_host(lxc['name']).get_vars())
-
-                # get QEMU vm's for this node
-                node_qemu_group = self.to_safe('%s%s' % (self.get_option('group_prefix'), ('%s_qemu' % node['node']).lower()))
-                self.inventory.add_group(node_qemu_group)
-                for qemu in self._get_qemu_per_node(node['node']):
-                    if qemu.get('template'):
-                        continue
+    def _can_add_host(self, name, properties):
+        '''Ensure that a host satisfies all defined hosts filters. If strict mode is
+        enabled, any error during host filter compositing will lead to an AnsibleError
+        being raised, otherwise the filter will be ignored.
+        '''
+        add_host = True
+        for host_filter in self.host_filters:
+            try:
+                add_host = self._compose(host_filter, properties)
```

Comment:
How about:
```suggestion
                if not self._compose(host_filter, properties):
                    return False
```
and removing the `add_host` variable?

Reply:
The problem with that fix is that `strict` mode would no longer fail if, in a sequence of filters, one of them returns `False` before the "bad" filter is reached. Something like this : 

```yaml
filters:
  - false
  - wup wuppity wup i love corn
```

would go through without problem, and I think it's wrong with regards to how `strict` should be interpreted.

I could fix it so that it returns directly if both `_compose()`'s result and `self.strict` are False. Do you think it would be worth it ?

Reply:
Well, your code currently does also not fail when this happens, since you have
```
            if not add_host:
                return False
```
*inside* the loop. To get the behavior you want, you have to remove that, and change the last line from `return True` to `return add_host`.

In any case, `strict` means that it should fail on errors (instead of ignoring them), it does *not* mean "make sure that every condition is actually evaluated to find out if there is an error" IMO.

Reply:
Oh ! Yep, I'd missed that, sorry.

OK, I'll change it the way you suggested.

Reply:
Done in commit a35518c